socorro/external/postgresql/connection_context.py

```diff
@@ -2,15 +2,32 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import os
 import socket
 import contextlib
 import psycopg2
 import psycopg2.extensions
+from urlparse import urlparse
 
-from configman.config_manager import RequiredConfig
-from configman import Namespace
+from configman import RequiredConfig, Namespace
+from configman.memoize import memoize
 
+#------------------------------------------------------------------------------
+# we're going to end up parsing the same url several times, there's no need to
+# actually do it more than once, so we wrap the method with a memoizer to speed
+# things along.
+@memoize()
+def parse_url(*args):
+    return urlparse(*args)
 
+#------------------------------------------------------------------------------
+def default_from_url(field, default):
+    if 'database_url' in os.environ:
```

Comment:
Why is only ever coming from the env? What if the database_url comes from an .ini file?
I like env variables and I'm sure it'll work but what if you put the database_url in the .ini or as a argument (e.g. `--database_url=postgres://...`)